database/kwdb.sql

```diff
@@ -0,0 +1,172 @@
+PRAGMA foreign_keys = 'ON';
+BEGIN TRANSACTION;
+-- Tables
+CREATE TABLE IF NOT EXISTS "command_label" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "name" TEXT NOT NULL UNIQUE,
+  PRIMARY KEY("id")
+);
+
+CREATE TABLE IF NOT EXISTS "config" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "name" TEXT NOT NULL,
+  "description" TEXT,
+  "path" TEXT NOT NULL UNIQUE,
+  PRIMARY KEY("id")
+);
+
+CREATE TABLE IF NOT EXISTS "event" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "date" TEXT NOT NULL,
+  "time" TEXT,
+  PRIMARY KEY("id")
+);
+
+CREATE TABLE IF NOT EXISTS "executed" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "command_label_id" INTEGER NOT NULL,
+  "elapsed_time_in_secs" INTEGER NOT NULL,
+  "status_id" INTEGER NOT NULL,
+  PRIMARY KEY("id"),
+  FOREIGN KEY("id") REFERENCES "event"("id")
+    ON UPDATE CASCADE ON DELETE CASCADE,
+  FOREIGN KEY("command_label_id") REFERENCES "command_label"("id")
+    ON UPDATE CASCADE ON DELETE RESTRICT,
+  FOREIGN KEY("status_id") REFERENCES "status"("id")
+    ON UPDATE CASCADE ON DELETE RESTRICT
+);
+
+CREATE TABLE IF NOT EXISTS "status" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "name" TEXT NOT NULL UNIQUE,
+  PRIMARY KEY("id")
+);
+
+CREATE TABLE IF NOT EXISTS "tag" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "name" TEXT NOT NULL UNIQUE,
+  PRIMARY KEY("id")
+);
+
+CREATE TABLE IF NOT EXISTS "timebox" (
+  "id" INTEGER NOT NULL UNIQUE,
+  "duration" INTEGER NOT NULL,
+  "description" TEXT,
+  "tag_id" INTEGER NOT NULL,
+  PRIMARY KEY("id"),
+  FOREIGN KEY("id") REFERENCES "event"("id")
+    ON UPDATE CASCADE ON DELETE CASCADE,
```

Comment:
This delete only happens if the event that the timebox references is deleted. This should only happen when clearing the database, I haven't implemented any mechanism to delete or alter the contents of the db directly. The way I see it the event table data shouldn't ever need to be changed. This is set to delete as well because the event is the "parent" to all, so deleting the event implies deleting the data related to it.

Reply:
But do we have plans to clean the database? In which scenario do we want to do that? I think it is easier to remove the current database and regenerate it.